test/base64_unittest.cc

```diff
@@ -0,0 +1,38 @@
+#include "../src/base64.h"
+#include "gtest/gtest.h"
+
+namespace {
+
+TEST(EncodeTest, EmptyEncode) {
+  const std::string empty_str("");
+  EXPECT_EQ(
+      "",
+      base64::Encode(empty_str)
+  );
+}
+
+TEST(EncodeTest, SimpleEncode) {
+  const std::string simple_str("tes");
+  EXPECT_EQ(
+      "dGVz",
+      base64::Encode(simple_str)
+  );
+}
+
+TEST(EncodeTest, NoPaddingEncode) {
+  const std::string simple_str("test0");
+  EXPECT_EQ(
+      "dGVzdDA",
+      base64::Encode(simple_str)
+  );
+}
+
+TEST(EncodeTest, NoDoublePaddingEncode) {
```

Comment:
Why "double padding"? Can you please elaborate?

Reply:
Sure: Typically this would have "==" as padding, hence the "double" as opposed to the above which would have "=" (single).  I guess its kind of a trivial case, happy to remove it.

Reply:
I think it's an important corner case and needs to be tested, but see my other note about test naming.

Reply:
Addressed above.